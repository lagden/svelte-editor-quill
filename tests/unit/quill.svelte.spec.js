import { setTimeout } from 'timers/promises'
import { beforeAll, expect, test } from 'vitest'
import { mount } from 'svelte'
import { Editor } from '$lib/index.js'

beforeAll(() => {
	document.body.innerHTML = '<main id="xxx"></main>'
})

function doc_query(selector) {
	const node = document.querySelector(selector)
	if (!node) {
		throw new Error(`No element found for selector: ${selector}`)
	}
	return node
}

test('Editor', async () => {
	const target = doc_query('main#xxx')
	const props = $state({
		options: {
			theme: 'snow',
			plainclipboard: true,
		},
	})
	const app = mount(Editor, { target, props })

	await setTimeout(3000)

	expect(app).toMatchSnapshot()

	// const div = doc_query('div.ql-editor')
	// expect(target).toBeDefined()
})
