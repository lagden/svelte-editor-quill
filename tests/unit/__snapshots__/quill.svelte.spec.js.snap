// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Editor 1`] = `
{
  "$destroy": [Function],
  "$on": [Function],
  "$set": [Function],
}
`;
